recipes/default.rb

```diff
@@ -46,6 +46,11 @@
   end
 end
 
+##  Set syncrepl_config dynamic values here
+node.default['openldap']['syncrepl_config']['searchbase'] = "\"#{node['openldap']['basedn']}\""
+node.default['openldap']['syncrepl_config']['binddn'] = "\"#{node['openldap']['syncrepl_cn']},#{node['openldap']['basedn']}\""
+node.default['openldap']['syncrepl_config']['credentials'] = "\"#{node['openldap']['slapd_replpw']}\""
```

Comment:
the code here is wrong though.  either you've misunderstood the problem i'm talking about, or you're trying to maintain backwards compat with broken code that needs fixing.  if anyone tries to set a default value on `node.default['openldap']['syncrepl_config']['searchbase']` it will get blown away.  if you don't intend that value to be 'settable' by users then it needs to not be an attribute.  if it is meant to be an attribute then you need to not always clobber someone's settings in recipe code.